pkg/kots/lint.go

```diff
@@ -344,6 +361,103 @@ func lintWithKubevalSchema(renderedFiles SpecFiles, originalFiles SpecFiles, sch
 	return lintExpressions, nil
 }
 
+func checkIfVersionExists(version string) (bool, error) {
+	url := "http://api.github.com/repos/replicatedhq/kots/commits/%s"
+	token := os.Getenv("GITHUB_API_TOKEN")
+	var bearer = "Bearer " + token
+
+	if !strings.HasPrefix(version, "v") {
+		version = "v" + version
+	}
+
+	rwMutex.RLock()
+	verIsCached := kotsVersions[version]
+	rwMutex.RUnlock()
+
+	if !verIsCached {
+		req, err := http.NewRequest("GET", fmt.Sprintf(url, version), nil)
+		if err != nil {
+			return false, errors.Wrap(err, "failed to create new request")
+		}
+		req.Header.Set("Authorization", bearer)
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if resp.StatusCode != 200 {
```

Comment:
we should check if it's 404 instead, fail and return an error if != 200 and != 404?. the version could exist but it's an internal github api issue that returns 500 status code for example.

Reply:
done